feat: expose `@slack/web-api` options to bolt apps

The `@slack/web-api` package has neat [options](https://tools.slack.dev/node-slack-sdk/reference/web-api/interfaces/WebClientOptions) for the `client`! Some of these are available to the top-level `App` constructor, but others are provided via `clientOptions`.

At the moment, `slackApiUrl` is [available](https://github.com/slackapi/bolt-js/blob/11be2fc85b24874711658d8ccde1a87836d85520/src/App.ts#L130) but not `allowAbsoluteUrls` or other `clientOptions`. #2292 is similar!

### Example

Following the updates of `@slack/web-api` in https://github.com/slackapi/bolt-js/pull/2473, the following is possible in app setups:

```js
const app = new App({
  token: process.env.SLACK_BOT_TOKEN,
  appToken: process.env.SLACK_APP_TOKEN,
  socketMode: true,
  logLevel: LogLevel.DEBUG,
  clientOptions: {
    allowAbsoluteUrls: false,
  },
});
```

Which can be tried with:

```js
app.client.apiCall("http://localhost:8080/", { /* ... */ });
```

But this will cause the `tsc` compiler to error:

```sh
error TS2353: Object literal may only specify known properties, and 'allowAbsoluteUrls' does not exist in type 'Pick<WebClientOptions, "slackApiUrl">'.
```

### Suggestion

I'm wondering if adding these types to `clientOptions` makes sense?:

- maxRequestConcurrency
- retryConfig
- timeout
- rejectRateLimitedCalls
- headers
- allowAbsoluteUrls
- attachOriginalToWebAPIRequestError
- requestInterceptor
- adapter

IMO the customizations possible are nice, but this might not align with adding options to `clientOptions` or `App` 🤔 

Another option might prefer to `Omit` certain options! This is an approach `@slack/oauth` takes:

https://github.com/slackapi/node-slack-sdk/blob/39b8992dc35f9fbf8d2afc0c00b3ba62326e8c88/packages/oauth/src/install-provider-options.ts#L98

That I think `@slack/bolt` might be setup well for already, using the same setup!: 🎁 ✨ 

https://github.com/slackapi/bolt-js/blob/11be2fc85b24874711658d8ccde1a87836d85520/src/App.ts#L336-L353

### Reference

Here's the current `WebClient` constructor options:

https://github.com/slackapi/node-slack-sdk/blob/39b8992dc35f9fbf8d2afc0c00b3ba62326e8c88/packages/web-api/src/WebClient.ts#L265-L284

Here's the current `App` constructor:

https://github.com/slackapi/bolt-js/blob/11be2fc85b24874711658d8ccde1a87836d85520/src/App.ts#L104-L137

### Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

--- a/src/App.ts
+++ b/src/App.ts
@@ -127,7 +127,12 @@ export interface AppOptions {
   logger?: Logger;
   logLevel?: LogLevel;
   ignoreSelf?: boolean;
-  clientOptions?: Pick<WebClientOptions, 'slackApiUrl'>;
+  /**
+   * Configurations for the web client used to send Slack API method requests.
+   *
+   * See {@link https://tools.slack.dev/node-slack-sdk/reference/web-api/interfaces/WebClientOptions} for more information.
+   */
+  clientOptions?: WebClientOptions;
   socketMode?: boolean;
   developerMode?: boolean;
   tokenVerificationEnabled?: boolean;

--- a/test/types/App.test-d.ts
+++ b/test/types/App.test-d.ts
@@ -1,0 +1,64 @@
+import { Agent } from 'node:http';
+import { ConsoleLogger, LogLevel } from '@slack/logger';
+import type { Installation, InstallationQuery } from '@slack/oauth';
+import { expectAssignable, expectError, expectType } from 'tsd';
+import App from '../../src/App';
+
+expectAssignable<App>(
+  new App({
+    token: 'xoxb-example',
+    signingSecret: 'secretpassword',
+  }),
+);
+
+expectAssignable<App>(
+  new App({
+    token: 'xoxb-example',
+    socketMode: true,
+    appToken: 'xapp-example',
+  }),
+);
+
+expectAssignable<App>(
+  new App({
+    logLevel: LogLevel.DEBUG,
+    signingSecret: 'secretpassword',
+    clientId: '0123456789',
+    clientSecret: 'randomvalues',
+    stateSecret: 'hidden',
+    scopes: ['chat:write'],
+    installationStore: {
+      storeInstallation: async <AuthVersion extends 'v1' | 'v2'>(installation: Installation<AuthVersion, boolean>) => {
+        expectType<Installation<AuthVersion, boolean>>(installation);
+      },
+      fetchInstallation: async (query) => {
+        expectType<InstallationQuery<boolean>>(query);
+        return {
+          user: { token: undefined, scopes: [], id: 'U0123456789' },
+          bot: { token: 'xoxb-example', scopes: ['chat:write'], id: 'B0123456789', userId: 'U0101010101' },
+          team: { id: 'T0123456789' },
+          enterprise: undefined,
+        };
+      },
+      deleteInstallation: async (query) => {
+        expectType<InstallationQuery<boolean>>(query);
+      },
+    },
+  }),
+);
+
+expectAssignable<App>(
+  new App({
+    clientOptions: {
+      agent: new Agent(),
+      allowAbsoluteUrls: false,
+      logger: new ConsoleLogger(),
+      retryConfig: {
+        forever: true,
+      },
+      slackApiUrl: 'http://localhost:8080/api/',
+    },
+  }),
+);
+
+expectError(new App({ password: 'randomish!' }));

